services/cd-service/pkg/service/overview_test.go

```diff
@@ -659,6 +660,183 @@ func TestOverviewService(t *testing.T) {
 	}
 }
 
+func TestGetApplicationDetails(t *testing.T) {
+	var dev = "dev"
+	var env = "development"
+	var appName = "test-app"
+	tcs := []struct {
+		Name             string
+		Setup            []repository.Transformer
+		Test             func(t *testing.T, svc *OverviewServiceServer, expectedResponse *api.GetAppDetailsResponse)
```

Comment:
this test function should be part of the actual tests, please avoid function pointers in test arrays